inc/packages/namespace.php

```diff
@@ -165,6 +166,35 @@ function fetch_metadata_doc( string $url ) {
 	return MetadataDocument::from_response( $response );
 }
 
+/**
+ * Fetch Metadata from local source.
+ *
+ * Solves issue where Metadata source is from same site.
+ * Mini-FAIR REST endpoint may time out under these circumstances.
+ *
+ * @param  bool|array $response Response from cache.
+ * @param  string $url URI for Metadata.
+ * @return bool|array
+ */
+function fetch_metadata_from_local( $response, $url ) {
+	if ( ! $response ) {
+		if ( str_contains( $url, home_url() ) ) {
+			$did = explode( '/', parse_url( $url, PHP_URL_PATH ) );
+			$did = array_pop( $did );
+			$body = wp_cache_get( 'fair-metadata-endpoint-' . $did, 'metadata-endpoints' );
+			$response  = [];
+			$response['headers'] = [];
+			$response['body'] = json_encode( $body );
+			$response = ! $body ? false : $response;
+			if ( $response ) {
+				wp_cache_set( CACHE_KEY . md5( $url ), $response, 'metadata-docs', CACHE_LIFETIME );
+			}
+		}
+	}
+
+	return $response;
+}
```

Comment:
With the above suggestion, `fetch_metadata_from_local()` could then become this:

```suggestion
 * @return array|WP_Error
 */
function fetch_metadata_from_local( $url ) {
	$did = explode( '/', parse_url( $url, PHP_URL_PATH ) );
	$did = array_pop( $did );
	
	$body = wp_cache_get( 'fair-metadata-endpoint-' . $did, 'metadata-endpoints' );
	if ( ! $body ) {
		return new WP_Error( 'fair.packages.metadata.failure', __( 'Local metadata not available.', 'fair' ) );
	}

	$response = [
		'headers' => [],
		'body' => json_encode( $body ),
	];
	wp_cache_set( CACHE_KEY . md5( $url ), $response, 'metadata-docs', CACHE_LIFETIME );
	return $response;
}
```
- Removed `$response` parameter.
- Removed `! $response` and `str_contains()` checks.
- Bailed with `WP_Error` when there's no metadata cached.
- Merged the `$response` array's generation into one block.